internal/cluster/cluster.go

```diff
@@ -1399,32 +1403,106 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	return m.setConnectivityMajorityGroupsForClusterInternal(cluster, db)
 }
 
-func (m *Manager) deleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) error {
-	log := logutil.FromContext(ctx, m.log)
+func (m *Manager) extractManifestFolderAndFilename(ctx context.Context, path string) (string, string, bool) {
+	parts := strings.Split(path, "/")
+	partsLength := len(parts)
+	if partsLength != 4 || parts[1] != "manifests" {
+		return "", "", false
+	}
+	folder := parts[2]
+	filename := parts[3]
+	return folder, filename, true
+}
+
+func (m *Manager) getClusterFilesForFolder(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) ([]string, error) {
 	path := filepath.Join(string(*c.ID), folder) + "/"
 	files, err := objectHandler.ListObjectsByPrefix(ctx, path)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to list files in %s", path)
 		m.log.WithError(err).Errorf(msg)
-		return common.NewApiError(
+		return nil, common.NewApiError(
 			http.StatusInternalServerError,
 			errors.Errorf(msg))
 	}
+	return files, err
+}
 
-	var failedToDelete []string
+func (m *Manager) ResetClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
```

Comment:
This function is only dealing with manifests, right? Maybe we should name it that way.